src/gateways/createArchiveLoader.ts

```diff
@@ -44,53 +44,55 @@ const defaultOptions = () => ({
   pageCache: createCache<string, ArchivePage>({maxRecords: 20}),
 });
 
-export default (archivePath: string, options: Options = {}) => {
+export default (getArchivePath: () => string, options: Options = {}) => {
   const {pageCache, bookCache, appPrefix, archivePrefix, disablePerBookPinning} = {
     ...defaultOptions(),
     ...options,
   };
 
-  const contentUrlBase = (host: string, bookId: string) => disablePerBookPinning
+  const contentUrlBase = (host: string, archivePath: string, bookId: string) => disablePerBookPinning
     ? `${host}${archivePath}`
```

Comment:
you don't have to pass it all around, you can just call it here, so that it gets called at the time of usage, that way it'll get the most recent value

Reply:
I've updated this